PoGo.NecroBot.Logic/Inventory.cs

```diff
@@ -643,6 +643,25 @@ public bool CanTransferPokemon(PokemonData pokemon)
 
             return true;
         }
+        
+        public int GetCandyToEvolve(PokemonSettings settings, EvolveFilter appliedFilter = null)
+        {
+            EvolutionBranch branch;
+            PokemonId evolveTo = PokemonId.Missingno;
+            if (appliedFilter != null && !string.IsNullOrEmpty(appliedFilter.EvolveTo) && Enum.TryParse<PokemonId>(appliedFilter.EvolveTo, true, out evolveTo))
+            {
+                branch = settings.EvolutionBranch.FirstOrDefault(x => x.Evolution == evolveTo);
+            }
+            else
+            {
+                branch = settings.EvolutionBranch.FirstOrDefault(x => x.EvolutionItemRequirement == ItemId.ItemUnknown);
```

Comment:
I think it is safer to check evolution item just in case branch[0] has item requirement.